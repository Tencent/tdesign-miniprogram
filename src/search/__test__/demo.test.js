/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import path from 'path';
import simulate from 'miniprogram-simulate';

const mapper = ['action', 'base', 'maxlength', 'other', 'shape'];

describe('Search', () => {
  mapper.forEach((demoName) => {
    it(`Search ${demoName} demo works fine`, () => {
      const id = load(path.resolve(__dirname, `../../search/_example/${demoName}/index`), demoName);
      const container = simulate.render(id);
      container.attach(document.createElement('parent-wrapper'));
      expect(container.toJSON()).toMatchSnapshot();
    });
  });
});
