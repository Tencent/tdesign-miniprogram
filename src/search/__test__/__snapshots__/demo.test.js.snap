// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search action demo works fine 1`] = `
<action>
  <wx-view
    class="example-search"
  >
    <t-search
      action=""
      center="{{true}}"
      placeholder="搜索预设文案"
      value=""
      bind:action-click="actionHandle"
      bind:blur="blurHandle"
      bind:change="changeHandle"
      bind:focus="focusHandle"
    />
  </wx-view>
</action>
`;

exports[`Search Search base demo works fine 1`] = `
<base>
  <wx-view
    class="example-search"
  >
    <t-search
      placeholder="搜索预设文案"
      value=""
    />
  </wx-view>
</base>
`;

exports[`Search Search maxlength demo works fine 1`] = `
<maxlength>
  <wx-view
    class="example-search"
  >
    <t-search
      maxlength="{{10}}"
      placeholder="最大输入10个字符"
    />
  </wx-view>
  <wx-view
    class="example-search"
  >
    <t-search
      maxcharacter="{{10}}"
      placeholder="最大输入10个字符，汉字算两个"
    />
  </wx-view>
</maxlength>
`;

exports[`Search Search other demo works fine 1`] = `
<other>
  <wx-view
    class="example-search"
  >
    <t-search
      center="{{true}}"
      placeholder="搜索预设文案"
      value=""
      bind:change="onChange"
    />
  </wx-view>
</other>
`;

exports[`Search Search shape demo works fine 1`] = `
<shape>
  <wx-view
    class="example-search"
  >
    <t-search
      placeholder="搜索预设文案"
    />
  </wx-view>
  <wx-view
    class="example-search"
  >
    <t-search
      placeholder="搜索预设文案"
      shape="round"
    />
  </wx-view>
</shape>
`;
